src/utilops.cc

```diff
@@ -3286,50 +3285,31 @@ void file_util_copy_path_to_clipboard(FileData *fd, gboolean quoted, gboolean ac
  */
 void file_util_path_list_to_clipboard(GList *fd_list, gboolean quoted, gboolean action_copy)
 {
-	ClipboardData *cbd;
-	FileData *fd;
-	GList *work;
-
-	if (options->clipboard_selection == CLIPBOARD_PRIMARY || options->clipboard_selection == CLIPBOARD_BOTH)
-		{
-		cbd = g_new0(ClipboardData, 1);
-		cbd->path_list = nullptr;
-		cbd->quoted = quoted;
-		cbd->action_copy = action_copy;
-		work = fd_list;
+	// FIXME Is it safe to use FileList::to_path_list()?
+	static const auto get_path_list = [](GList *fd_list)
+	{
+		GList *path_list = nullptr;
 
-		while (work)
+		for (GList *work = fd_list; work; work = work->next)
 			{
-			fd = static_cast<FileData *>(work->data);
-			work = work->next;
+			auto *fd = static_cast<FileData *>(work->data);
 
 			if (!fd || !*fd->path) continue;
```

Comment:
@xsdg Does it make sense to add this condition to `FileList::to_path_list()`?

Reply:
Probably not without checking the assumptions from the callsites.  In particular, I can imagine a caller assuming that the returned list will have the same length as the list that was passed in, which could cause some subtle bugs if that is violated.

On the upside, the function is only called from two places (`bar-sort.cc` and `uri-utils.cc`).  Unfortunately, the usage in `uri-utils.cc` (in `uris_from_filelist`) is used in a bunch of different places, and seems to be related to selection logic.